src/modules/iotjs_module_mqtt.c

```diff
@@ -349,6 +380,138 @@ JS_FUNCTION(MqttPublish) {
 }
 
 
+JS_FUNCTION(MqttReceive) {
+  DJS_CHECK_THIS();
+  DJS_CHECK_ARGS(1, object);
+
+  jerry_value_t jnat = JS_GET_ARG(0, object);
+
+  void *native_p;
+  JNativeInfoType *out_native_info;
+  bool has_p =
+      jerry_get_object_native_pointer(jnat, &native_p, &out_native_info);
+  if (!has_p || out_native_info != &mqttclient_native_info) {
+    return JS_CREATE_ERROR(COMMON, "MQTT native pointer not available");
+  }
+  iotjs_mqttclient_t *mqttclient = (iotjs_mqttclient_t *)native_p;
+
+  jerry_value_t jbuffer = JS_GET_ARG(1, object);
+  iotjs_bufferwrap_t *buff_recv = iotjs_bufferwrap_from_jbuffer(jbuffer);
+  IOTJS_ASSERT(buff_recv->length != 0);
+
+  uint8_t first_byte;
+  char *tmp_ptr;
+  if (!mqttclient->buffer_length) {
+    // The local buffer is empty therefore either full packages arrived
+    // or this is the first packet.
+    first_byte = (buff_recv->buffer[0] >> 4) & 0x0F;
+    tmp_ptr = buff_recv->buffer;
```

Comment:
What happens if mqttclient->buffer only contains a single byte?

I would see the following options:
1) Always do concatenation first.
2) Create a byte reader function, which virtually concatenates the two buffers, and returns a byte provided by an index.

The 2nd uses less memory, but the 1st is easier to implement. Which one you prefer?